### ISSUE TITLE
Defaults are missing in the namespace of the sub subcommand

### ISSUE THREAD

**Issue by obaranov** [omni-us/jsonargparse#29]
Hi, 

I've been trying to design cli with the several levels of subcommands. I want something like that:

`app subcommand1 subcommand2 --arg1=foo`

The problem is that when I add an argument with the default value into the subcommand2, this argument is not getting populated in the subcommand2 namespace. 

Here is my code to verify that: 

```python
parser = ArgumentParser()
subcommand = ArgumentParser()
# this is a default argument for subcommand
subcommand.add_argument('--sub-foo', default="bar")

subcommand_subcommand = ArgumentParser()
# this is a default argument for subcommand2. This one will not be populated!
subcommand_subcommand.add_argument('--sub-sub-foo', default="bar")

root_subcommands = parser.add_subcommands()
root_subcommands.add_subcommand('subcommand1', subcommand)

sub_subcommands = subcommand.add_subcommands()
sub_subcommands.add_subcommand("subcommand2", subcommand_subcommand)


args = parser.parse_args(['subcommand1', 'subcommand2'])
print(args)
```

As the result I'm getting that:

`Namespace(__cwd__=['....'], subcommand='subcommand1', subcommand1=Namespace(sub_foo='bar', subcommand='subcommand2'))`

You can see here that the `subcommand2` Namespace is missing in the result. For the  `subcommand1` namespace you see that the default argument is present (sub_foo='bar'). 

Is that a bug, or am I doing something wrong here?

Best regards, 
Oleksii

**Comment by mauvilsa** [comment]
> Is that a bug, or am I doing something wrong here?

I had not considered the case of several levels of subcommands. Not a bug, simply not yet implemented. Certainly makes sense that several levels of subcommands works as you expected. I will look into it to figure out how difficult it would be to implement.

**Comment by mauvilsa** [comment]
I have fixed the issue with the defaults. Though I don't yet consider this to be an officially supported feature. Thus for the time being  I will not create a new release of the package. Please clone the repo to try it out.

As a side note. The subcommands should be added in the order of the levels, just like in your example. First `subcommand1` and then `subcommand2`. If they are added in a different order, there wouldn't be any error, but some things don't work as expected, in particular parsing environment variables and the help messages.

**Comment by obaranov** [comment]
Works fine for me now. Thank you for the quick fix :)

**Comment by mauvilsa** [comment]
This is now included in release v3.1.0.
